04_predict_controls/nbconverted/plot_msi_results.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot microsatellite instability classification results
+# 
+# Here, we'll visualize the results of the MSI prediction experiments. Remember that we're trying to classify MSI-H ("high" microsatellite instability, inferred from sequencing assays) against all other classes (MSI-L = "low" microsatellite instability, MSS = "microsatellite stable" tumors, "indeterminate" tumors).
+# 
+# Results analyzed here are generated by the `run_msi_prediction.py` script.
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+import mpmp.utilities.data_utilities as du
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+
+
+# ### Load results
+
+# In[2]:
+
+
+# set results directory
+results_dir = Path(cfg.results_dirs['msi'], 'msi_5000_top_mad_std').resolve()
+
+
+# In[3]:
+
+
+results_df = au.load_msi_results(results_dir)
+
+print(results_df.shape)
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# ### Plot results per cancer type, on true labels
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (14, 12)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(2, 1)
+
+plot_df = (
+    results_df[(results_df.data_type == 'test') &
+               (results_df.signal == 'signal')]
+      .sort_values(by=['signal', 'training_data'])
+)
+
+sns.boxplot(data=plot_df, x='identifier', y='aupr', hue='training_data', ax=axarr[0])
+axarr[0].set_title('MSI prediction performance, AUPR')
+axarr[0].set_xlabel('Cancer type')
+axarr[0].set_ylabel('AUPR')
+axarr[0].set_ylim(0.35, 1.05)
+axarr[0].legend(title='Data type')
+
+sns.boxplot(data=plot_df, x='identifier', y='auroc', hue='training_data', ax=axarr[1])
+axarr[1].set_title('MSI prediction performance, AUROC')
+axarr[1].set_xlabel('Cancer type')
+axarr[1].set_ylabel('AUROC')
+axarr[1].set_ylim(0.35, 1.05)
+axarr[1].legend(title='Data type')
+
+
+# ### Plot results per cancer type, corrected for permuted labels baseline
```

Comment:
What do you mean here that you corrected for permuted labels baseline?

Reply:
In most of our mutation prediction experiments, instead of directly reporting classification metrics (AUPR, AUROC etc), we report the difference in these metrics vs. a model where the labels are permuted. The idea behind this is that the permuted-labels models are a simple way to correct for differences in the baseline frequency of different mutations (or in this case, in the baseline frequency of MSI status across different cancer types).

For example, in cases where a gene is very frequently mutated, the permuted labels baseline will do well just by chance (i.e. just by predicting "mutation" every time), so just looking at raw classification metrics may give an inflated score for the true predictor in those cases. If the gene is mutated around 50% of the time (balanced labels) this is less of an issue, but in many cases that's not true for us, unfortunately.

I'll add a note here about this, so the notebook will make more sense on its own (we have a discussion of this in the methods section of our paper, and I think I made notes in some of the mutation prediction notebooks too).

Reply:
That makes sense thanks for clarifying